notebook/services/contents/fileio.py

```diff
@@ -124,6 +118,34 @@ def atomic_writing(path, text=True, encoding='utf-8', log=None, **kwargs):
         os.remove(tmp_path)
 
 
+def open_file(path, text=True, encoding='utf-8', log=None, **kwargs):
+    """Open file for contextmanager for writing
+
+    Parameters
+    ----------
+    path : str
+      The target file to write to.
+
+    text : bool, optional
+      Whether to open the file in text mode (i.e. to write unicode). Default is
+      True.
+
+    encoding : str, optional
+      The encoding to use for files opened in text mode. Default is UTF-8.
+
+    **kwargs
+      Passed to :func:`io.open`."""
+
+    if os.path.islink(path):
+        path = os.path.join(os.path.dirname(path), os.readlink(path))
```

Comment:
This `if` block shouldn't be part of the reduction since this same code exists in [`atomic_writting()`](https://github.com/jupyter/notebook/pull/5278/files#diff-81cc8fe92165832b275b72c7a4df05b0L94-L95) with dependent code in between.